### ISSUE TITLE
Should `useCodeBlocks` output valid TypeScript?

### ISSUE THREAD

**Issue by lukekarrys** [typedoc2md/typedoc-plugin-markdown#741]
### What package is your question related to?

typedoc-plugin-markdown

### What would you like to ask?

I have a project that is using `prettier` to format all the markdown generated by `typedoc-plugin-markdown`. (Not sure if this is relevant but it is using `remarkPlugins: ['unified-prettier']` currently but I also tested with the `formatWithPrettier` option which I plan to migrate to.)

The reason for my question is that I was searching issues and found this comment https://github.com/typedoc2md/typedoc-plugin-markdown/issues/580#issuecomment-1964926403 indicating that at least for functions `useCodeBlocks: true` will change the output so it is valid TypeScript that prettier can then format.

### Question

Is the ability to format a declaration or signature a goal for the output of `typedoc-plugin-markdown` when `useCodeBlocks` is turned on?

### Example

Currently this plugin does a good job at formatting for readability by inserting newlines when `useCodeBlocks` is turned on. I have a (edge?) case that doesn't get formatted to multiple lines and it would be nice to rely on prettier for this behavior.

This specific case could be fixed by inserting newlines, but I'm more interested in whether I can leverage the rest of the features prettier has, which only work if the output is valid TypeScript.

**TypeScript source**
```ts
export type DepID =
  | `${'' | 'git'}${Delimiter}${string}${Delimiter}${string}${Delimiter}${string}`
  | `${'' | 'git'}${Delimiter}${string}${Delimiter}${string}`
  | `${'file' | 'remote' | 'workspace'}${Delimiter}${string}${Delimiter}${string}`
  | `${'file' | 'remote' | 'workspace'}${Delimiter}${string}`

export type DepIDTuple =
  | [type: 'git', string, string, string]
  | [type: 'registry', string, string, string]
  | [type: 'file', string, string]
  | [type: 'remote', string, string]
  | [type: 'workspace', string, string]
```

**Markdown Output**

Here's a [playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBidACAInADgSSwB0oSADCmAT1zgGcwAnAS1xhOtuz0KQzIAkwIiBEYAPhhEBzZuxABfITgA2zALZy4jJcDowWUabtUatOofsPHlcNZvgW9B5kYVkJ-ISLGSZckRM7M0ddK1cbYFMHbTCXNw9JQWEQADNmFThfKRBGOHUIeGyRAHcIRgBrOlwAQzAsxVt7czjrIObQy3jjRK8U9Mzi3PzChs9S8qra+sCmkNiutrIAbnIKEhJ0VG4CLAAVAFdcQdIoCjIYOg4aOB3CQ+O4JBIMTwBtf3kAGgxwox+-tIAd0ALovd4iPKyKxUETA6zwiKIoxgqCvSQfNIZBrIoG-UHgjGQkZFEC43Go9EYTFlSrVOo4-EIpkREGrM4UMkgCBsZjQOjIUA1RiMCAlAAKwoQApQNRUJRqVAFXxAACNGHUKnAYABlGrqOAAGVccGQqTldDgKvVmu1OumEWQBgOVpAls0TsYLpVcAAHrQWAbYHK9tooMLmPQzRbXXQIpkAIoHUbRlSWlUAKzovp18bgSZTSHNaddAEdk-BxaLcDKQDU6ABaKBwOAAEzbXIMNQyEQAwhB1Ooasg6yoVFy40ZMgBBGAuVUHSvaY3N1PpkAACxg6hUAHUN1p6fUddK5MwAG5yKgjsB0ZUgc8u-BQduwHVMVgwacvnXUTJr11cFFS1dw1XARyA+htHPU0VVcS1GBgKsamkIcAJVWpGAQkch0qVsxSgLkgNcGBd2YVsYA3ZAAA4AAYVTyctmDyZDUOHIsYxVGAalVMiKKopAACYVQOS09h4mVi3XfJVTbdtW0NGojAOFC4AAMXKIc50dWVFwgRQFCAA) which can be edited to replace `:` with `=` and then format the code blocks.

````
### DepID

```ts
type DepID: `${"" | "git"}${Delimiter}${string}${Delimiter}${string}${Delimiter}${string}` | `${"" | "git"}${Delimiter}${string}${Delimiter}${string}` | `${"file" | "remote" | "workspace"}${Delimiter}${string}${Delimiter}${string}` | `${"file" | "remote" | "workspace"}${Delimiter}${string}`;
```

### DepIDTuple

```ts
type DepIDTuple:
  | ["git", string, string, string]
  | ["registry", string, string, string]
  | ["file", string, string]
  | ["remote", string, string]
  | ["workspace", string, string];
```
````

**Comment by tgreyuk** [comment]
@lukekarrys thanks.  

`typedoc-plugin-markdown@4.3.3` will now render type aliases inside code blocks correctly with `=`.

`:` is only valid for type annotations for variables and function signatures etc.

**Comment by lukekarrys** [comment]
Thanks for the quick fix @tgreyuk! I deployed the v4.3.3 and it works great.

Going a bit deeper on my overall question, there are obviously some rendered code blocks that cannot be made into valid TypeScript. For example:

```ts
optional type: EdgeLike["type"];
```

Maybe I'll write a Prettier parser that can figure out what to do in those cases (and use the `typedoc-plugin-markdown` snapshots as a test suite 😂). That would be a lot of work to make semicolons and double vs single quotes configurable, but I've never known a rabbit hole too deep.

Thanks again! I'll close this as my question has been answered.
